kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/RulesExecutionService.java

```diff
@@ -52,8 +55,17 @@ public ExecutionResults call(KieContainerInstance kci, BatchExecutionCommand exe
             if( command.getLookup() != null ) {
                 ks = context.getKieSessionLookupManager().lookup(command.getLookup(), kci, context);
             } else {
-                // if no session ID is defined, then the default is a stateful session
-                ks = ((KieContainerInstanceImpl)kci).getKieContainer().getKieSession();
+                // if no session ID is defined, then use default ksession. Stateful session takes priority
+                KieProject kieProject = ((KieContainerImpl)kci.getKieContainer()).getKieProject();
+                KieSessionModel defaultStatefulModel = kieProject.getDefaultKieSession();
+                KieSessionModel defaultStatelessModel = kieProject.getDefaultStatelessKieSession();
+                if (defaultStatefulModel != null) {
+                    ks = ((KieContainerInstanceImpl)kci).getKieContainer().getKieSession();
+                } else if (defaultStatelessModel != null) {
+                    ks = ((KieContainerInstanceImpl)kci).getKieContainer().getStatelessKieSession();
+                } else {
+                    throw new IllegalStateException("No default KieSession found on container '" + kci.getContainerId() + "'.");
+                }
```

Comment:
@tkobayas It seems to me that you had to apply exactly the same fix in 2 different places. Is there a reasonable way to avoid this code duplication?

Reply:
@mariofusco Ahh, yes. You are right. I'm going to refactor it.